tests/unit/locking_test.cc

```diff
@@ -420,3 +422,170 @@ SEASTAR_THREAD_TEST_CASE(test_with_lock_typed_return_throwing_move_func) {
         }
     }
 }
+
+SEASTAR_TEST_CASE(test_shared_mutex_locks) {
+    shared_mutex sm;
+
+    {
+        const auto ulock = co_await get_unique_lock(sm);
```

Comment:
An idea: wouldn't it be more "OO-like" and easier to use if you also added a method, sm.get_unique_lock() (or maybe name it differently, to be more like the rest of the Seastar API)?
Of course it will be implemented using the get_unique_lock() template, but it will still be useful to have methods. Or so I think.

Reply:
I think that's a tricky case because on one hand I agree that it would be more convenient and natural to have it as a method, but on the other there are two arguments against it:
1. code duplication
2. inconsistency with Seastar (e.g. semaphores don't have an equivalent method of `seastar::get_units`) and STL (neither `std::mutex`, nor `std::shared_mutex` has a method returning a lock. Though it might be because of point 1)

Reply:
Ok. We can always add more methods later if we want.